gslib/command_runner.py

```diff
@@ -424,6 +426,48 @@ def RunNamedCommand(self,
           )))
     return return_code
 
+  def SkipUpdateCheck(self, command_name):
+    """Helper function that will determine if update checks should be skipped.
+
+    Args:
+      command_name: The name of the command being run.
+
+    Returns:
+      True if:
+      - gsutil is not connected to a tty (e.g., if being run from cron);
+      - user is running gsutil -q
+      - user specified gs_host (which could be a non-production different
+        service instance, in which case credentials won't work for checking
+        gsutil tarball)."""
+    logger = logging.getLogger()
+    if (not system_util.IsRunningInteractively() or
+        not logger.isEnabledFor(logging.INFO) or
+        boto_util.HasUserSpecifiedGsHost()):
+      return True
+    return False
+
+  def MaybePromptForPythonUpdate(self, command_name):
+    """Alert the user that they should install Python 3.
+
+    Args:
+      command_name: The name of the command being run.
+
+    Returns:
+      True if a prompt was output.
+    """
+    logger = logging.getLogger()
+    if (self.SkipUpdateCheck(command_name) or
+        command_name not in ('update', 'ver', 'version') or
+        boto.config.getbool('GSUtil', 'skip_python_update_prompt', False) or
+        sys.version_info.major != 2):
+      return False
+
+    # Notify the user about Python 2 deprecation.
+    print_to_fd(
+        'Gsutil 5 drops Python 2 support. Please install Python 3 to update '
+        'to the latest version of gsutil. https://goo.gle/py3\n')
```

Comment:
Should we add a message here about how they can turn off this warning?

Reply:
Knowing this is only used on `version` and `update` now i don't think the config option is worth documenting, although i still think it's good to have.

Reply:
Ah right, I totally missed that this is only for version and update command. In that case, I agree, that it is probably not worth documenting it.